controller/src/test/java/io/pravega/controller/store/stream/StreamTest.java

```diff
@@ -0,0 +1,129 @@
+        testStream(stream);
+    }
+
+    private void testStream(PersistentStreamBase<Integer> stream) throws InterruptedException, ExecutionException {
+        long creationTime1 = System.currentTimeMillis();
+        long creationTime2 = creationTime1 + 1;
+        final ScalingPolicy policy1 = ScalingPolicy.fixed(5);
+        final ScalingPolicy policy2 = ScalingPolicy.fixed(6);
+
+        final StreamConfiguration streamConfig1 = StreamConfiguration.builder().scope("test").streamName("test").scalingPolicy(policy1).build();
+        final StreamConfiguration streamConfig2 = StreamConfiguration.builder().scope("test").streamName("test").scalingPolicy(policy2).build();
+
+        CreateStreamResponse response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        stream.storeCreationTimeIfNotExists(creationTime1).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime2).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+
+        stream.createConfigurationIfNotExists(streamConfig1).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime2).get();
+        assertEquals(CreateStreamResponse.CreateStatus.EXISTS_CREATING, response.getStatus());
+
+        stream.createStateIfNotExists(State.UNKNOWN).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime2).get();
+        assertEquals(CreateStreamResponse.CreateStatus.EXISTS_CREATING, response.getStatus());
+
+        stream.updateState(State.CREATING).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
```

Comment:
I don't understand this behavior, we update the state of the stream to `CREATING` in line 101. Why should it make any difference to use a different config with respect to checking that stream exists? I also don't understand why we  get `EXISTS_CREATING` the second time, I'd `checkStreamExists` to not change state at all.